### PR TITLE
WIP: log to span exporter

### PR DESCRIPTION
**Pull request by gramidt**
**Description:** 
This exporter receives logs, converts them to trace spans, then exports the spans via OTLP over gRPC.

NOTE: This has been migrated from a PoC in a private distribution, and will be split up into multiple PRs once a more robust implementation is finalized. 

**Link to tracking Issue:**
#3071 

**Testing:**
Unit Tests

**Documentation:**
Code + README.

### CODE REVIEW

File: exporter/logtospanexporter/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- `key_file` (no default): path to the TLS key to use for TLS required connections. Should
+  only be used if `insecure` is set to false.
+
+Example:
+
+```yaml
+exporters:
+  logtospan:
+    endpoint: "otelcol:4317"
+    field_map:
```
Comment (clhain) [inline exporter/logtospanexporter/README.md:60]: Need TLS params or insecure=false here and in the example config files.

File: exporter/logtospanexporter/config.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	// TraceType related configuration
+	// Type of span that the log represents. (default w3c)
+	//   * w3c - Logs contain W3C Trace Context 'traceparent' and 'tracestate' attributes
+	TraceType TraceType `mapstructure:"trace_type"`
+
+	// W3CTraceContextFields defines the log fields used for the W3C Trace Context
+	W3CTraceContextFields W3CTraceContextFields `mapstructure:"w3c"`
+
+	// SpanName defines the name of the span.
+	SpanName string `mapstructure:"span_name"`
```
Comment (clhain) [inline exporter/logtospanexporter/config.go:30]: Would it make sense to treat SpanName like the others and derive it from a specified attribute (maybe with an optional default)?

Comment (gramidt) [reply exporter/logtospanexporter/config.go:30]: That's exactly what it is; however, the comment is incorrect. 

I also thought about having a default, but I'm not sure a default would ever be used...

Comment (clhain) [reply exporter/logtospanexporter/config.go:30]: haha sorry that was obvious from context I just didn't catch it. Agreed probably no need for default.

Comment (gramidt) [reply exporter/logtospanexporter/config.go:30]: No need to apologize. It helped catch the terrible comment :)

**Comment by jriguera** [comment]
It seems this PR got abandoned. We have an use case where access logs from a PaaS (Cloudfoundry) have the tracing headers like  `x_b3_traceid:"12cf43fd0b6842067e45edca9c881012" x_b3_spanid:"7e45edca9c881012" x_b3_parentspanid:"-" ` and we would like to process those log lines as spans.

**Pull request closed without merging**